src/crimson/osd/heartbeat.cc

```diff
@@ -309,6 +312,26 @@ seastar::future<> Heartbeat::handle_you_died()
   return seastar::now();
 }
 
+void Heartbeat::heartbeat_check()
+{
+  failure_queue_t failure_queue;
+  for (const auto& [osd, peer_info]: peers) {
+    if (clock::is_zero(peer_info.first_tx))
+      continue;
+
+    if (peer_info.is_unhealthy(clock::now())) {
+      logger().error(" heartbeat_check: no reply from osd.{} "
+		     "since back {} front {} (oldest deadline {})",
+		     osd, peer_info.last_rx_back, peer_info.last_rx_front,
+		     peer_info.ping_history.begin()->second.deadline);
+      failure_queue[osd] = std::min(peer_info.last_rx_back,
+					peer_info.last_rx_front);
+    }
+  }
+  if (!failure_queue.empty())
+    send_failures(std::move(failure_queue));
```

Comment:
if you are ignoring the returned future, please note down the reason. and to be explicit to ignore it.